File: typescript/easy/1-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
  let cache: Map<number, number> = new Map();

  for (let i = 0; i < nums.length; i++) {
      const val = nums[i];
      const complement = target - val;

      if (cache.has(complement)) {
          return [cache.get(complement)!, i]
      }

      cache.set(val, i);
  }
};

File: go/easy/1-two-sum.go
func twoSum(nums []int, target int) []int {
	cache := make(map[int]int)

	for i, val := range nums {
		if j, exists := cache[target-val]; exists {
			return []int{i, j}
		}

		cache[val] = i
	}

	return []int{}
}

